### ISSUE TITLE
SSAInvokeDynamic problem in 1.5.5 and 1.5.6-SNAPSHOT

### ISSUE THREAD

**Issue by MarcelBruch** [wala/WALA#783]
I'm trying to analyse [ru.saidgadjiev.ormnext:ormnext-core::jar:1.2](https://mvnrepository.com/artifact/ru.saidgadjiev.ormnext/ormnext-core/1.2) 

While analysing ru.saidgadjiev.ormnext.core.field.FieldAccessor.makeGetter(java.lang.invoke.MethodHandles$Lookup, java.lang.invoke.MethodHandle) Wala 1.5.5 fails with an error that the given method is not an SSAInvokeDynamicInstruction. 

Wala 1.5.6-SNAPSHOT does not produce an error (it's a normal SSAInvokeInstruction) but I because of that I don't have access to the Bootstrap object and thus cannot resolve the lambda method.

Dependencies used: log4j/log4j/1.2.17/log4j-1.2.17.jar, ru/saidgadjiev/ormnext/proxy-maker/1.2/proxy-maker-1.2.jar and an jdk11.

**Comment by msridhar** [comment]
Two things.  When I look at the source code for `makeGetter` I see:
```java
    /**
     * Map method handle to {@link Function}.
     *
     * @param lookup target lookup
     * @param handle target handle
     * @return function
     * @throws Throwable any exceptions
     */
    private static Function<Object, Object> makeGetter(MethodHandles.Lookup lookup, MethodHandle handle)
            throws Throwable {
        return (Function<Object, Object>) LambdaMetafactory.metafactory(
                lookup,
                "apply",
                MethodType.methodType(Function.class),
                MethodType.methodType(Object.class, Object.class),
                handle,
                handle.type()
        ).getTarget().invokeExact();
    }
```
This is not a lambda, but a direct usage of the metafactory.  We only support `invokedynamic` as used in lambdas in WALA for now.  We could look into supporting this case if it's easy.

Next, I was trying to look at the bytecodes for this method to see the `invokedynamic`, but when I download [the jar file](https://repo1.maven.org/maven2/ru/saidgadjiev/ormnext/ormnext-core/1.2/ormnext-core-1.2.jar), the bytecodes I see in the `FieldAccessor` class are very different from the source code and the `makeGetter` method is not present at all.  Are you compiling this code from source?

**Comment by MarcelBruch** [comment]
OMG... my apologies! I was just looking at the errors produced in 1.5.4 and 1.5.5 and concluded that there was a problem in WALA :-(  Next time I'll look at the source code in question before creating an issue...

**Comment by msridhar** [comment]
@MarcelBruch no problem! Should I create a 1.5.6 release so you can get the latest crash fix that is only on the master branch?

**Comment by MarcelBruch** [comment]
It depends on how stable the snapshot is :-)
At the moment I'm fine using the snapshot from the sonatype repository. 

I'm going to run a large set of analyses next week and will report issues I might experience with 1.5.6-SNAPSHOT. After that a release would be fine.

**Comment by msridhar** [comment]
Sounds good.  The snapshot only has minor bug fixes so it should be as stable as 1.5.5.  But yes, please let us know how the large set of analyses goes!
